lib/drivers/bayerContourNext.js

```diff
@@ -392,18 +392,27 @@ module.exports = function (config) {
   };
 
   var getASTMMessage = function (timeout, retries, callback) {
+    var e = new Error('Timeout error.');
+    e.name = 'TIMEOUT';
+
     var abortTimer = setTimeout(function () {
       clearInterval(listenTimer);
       debug('TIMEOUT');
-      var e = new Error('Timeout error.');
-      e.name = 'TIMEOUT';
       return callback(e, null);
     }, timeout);
 
     var listenTimer = setInterval(function () {
       hidDevice.receive(function(raw) {
         var packet = new Uint8Array(raw);
 
+        // we need to reset the timeout
+        clearTimeout(abortTimer);
+        abortTimer = setTimeout(function () {
```

Comment:
Might want to shove this in a function to DRY this up. And you can pull the `Error` back into the function since it will only be there once.